### ISSUE TITLE
function-argument shadowing

### ISSUE THREAD

**Issue by glenjamin** [eslint/eslint#1584]
From the docs: http://eslint.org/docs/rules/no-shadow.html

> The following patterns are considered okay and do not cause warnings:
> 
> ``` js
> var a = 3;
> function b(a) {
>     a = 10;
> }
> b(a);
> ```

Would it be possible to include this check as an option? I just lost a bunch of time trying to find a bug which was caused by this - I had assumed it wasn't due to shadowing because I knew the linter was checking for it!

I can try and put together a PR if that's easier?
